common/djangoapps/entitlements/models.py

```diff
@@ -301,3 +301,33 @@ def get_active_entitlements_for_user(cls, user):
             expired_at__isnull=False,
             enrollment_course_run=None
         ).select_related('user').select_related('enrollment_course_run')
+
+
+class CourseEntitlementSupportDetail(TimeStampedModel):
+    """
+    Table recording support interactions with an entitlement
+    """
+    LEAVESESSION = 'LS'
+    CHANGESESSION = 'CS'
+    LEARNER_REQUEST_NEW = 'RN'
+    COURSE_TEAM_REQUEST_NEW = 'CN'
+    OTHER = 'OT'
+    ENTITLEMENT_SUPPORT_REASONS = (
+        (LEAVESESSION, u'Learner requested leave session for expired entitlement'),
+        (CHANGESESSION, u'Learner requested session change for expired entitlement'),
+        (LEARNER_REQUEST_NEW, u'Learner requested new entitlement'),
+        (COURSE_TEAM_REQUEST_NEW, u'Course team requested entitlement for learnerg'),
+        (OTHER, u'Other'),
+    )
+    entitlement = models.ForeignKey('entitlements.CourseEntitlement')
+    support_user = models.ForeignKey(settings.AUTH_USER_MODEL)
+
+    reason = models.CharField(max_length=1, choices=ENTITLEMENT_SUPPORT_REASONS)
```

Comment:
shouldn't this max length be 2?

Reply:
Good point, I've bumped it up to 15 to match the more verbose values we are using now.